pkg/transform/mapper.go

```diff
@@ -12,176 +12,312 @@
 package transform
 
 import (
+	"context"
 	"encoding/json"
-	"errors"
 	"fmt"
-	"log"
+	"time"
 
 	"github.com/itchyny/gojq"
+
+	"github.com/snowplow/snowbridge/config"
 	"github.com/snowplow/snowbridge/pkg/models"
 )
 
-// This works perfectly, is a million times simpler to implement, and prevents us from being blocked in future if we haven't predicted a requirement
-var examplePureJQConfig = `{
-	field1: .app_id,
-	field2: { field2: .contexts_com_acme_just_ints_1[0] },
-	fieldWithOtherCoalesceExample: ( .app_id // .contexts_com_acme_just_ints_1[0] ),
-	manualUnnest: { just_ints_integerField: .contexts_com_acme_just_ints_1[0].integerField },
-	arraySpecified: [ .app_id, .event_id ]
-  }`
-
-func grabFromGenericJQConfig(inputData map[string]any, config string) []byte {
-	query, err := gojq.Parse(config)
-	if err != nil {
-		panic(err)
-	}
+// JQMapperConfig represents the configuration for the JQ transformation
+type JQMapperConfig struct {
+	JQCommand  string `hcl:"jq_command"`
+	RunTimeout int    `hcl:"timeout_sec,optional"`
+	SpMode     bool   `hcl:"snowplow_mode,optional"`
```

Comment:
For JS transformations, `spMode` has actually made things more complicated than they needed to be. I wonder if we need it here - I think likely not.

JQ is a JSON mapping tool, I think it's actually simpler for the user to understand that the input needs to be JSON, and for snowplow data there's a transformation to turn the data into JSON.

Reply:
Actually never mind - I forgot that the input is a map